pypdf/generic/_base.py

```diff
@@ -589,7 +594,7 @@ def write_to_stream(
     def renumber(self) -> bytes:
         out = self[0].encode("utf-8")
         if out != b"/":
-            logger_warning(f"Incorrect first char in NameObject:({self})", __name__)
+            raise PyPdfError(f"Incorrect first char in NameObject:({self})", __name__)
```

Comment:
This is a breaking change and should follow our deprecation process: https://pypdf.readthedocs.io/en/latest/dev/deprecations.html